installers/install.sh

```diff
@@ -112,8 +112,18 @@ mkdir -p "$INSTALLERTMPDIR"
 
 if [ -z "$VERSION" ]; then
   # Determine the latest version to fetch.
-  STATEURL="$BASE_INFO_URL?channel=$CHANNEL&source=install&platform=$OS"
-  $FETCH $INSTALLERTMPDIR/info.json $STATEURL || exit 1
+  JSONURL="$BASE_INFO_URL?channel=$CHANNEL&source=install&platform=$OS"
+elif [ -z "`echo $VERSION | grep -o '\-SHA'`" ]; then
+  # Determine the full version SHA to fetch.
+  JSONURL="$BASE_INFO_URL?channel=$CHANNEL&source=install&platform=$OS&target-version=$VERSION"
+fi
+
+if [ ! -z "$JSONURL" ]; then
```

Comment:
Could you add comments on the conditionals to explain what they are? I find these take me a while to interpret.

Reply:
Good idea.